src/Microsoft.Health.Fhir.Liquid.Converter/Processors/JsonToHl7v2Processor.cs

```diff
@@ -0,0 +1,179 @@
+using Microsoft.Health.Fhir.Liquid.Converter.Models.Json;
+using Microsoft.Health.Fhir.Liquid.Converter.Parsers;
+using Newtonsoft.Json.Linq;
+using NJsonSchema;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter.Processors
+{
+    public class JsonToHl7v2Processor : BaseProcessor
+    {
+        private readonly IDataParser _parser = new JsonDataParser();
+
+        private string[] _segmentsWithFieldSeparator = new string[] { "MSH", "BHS", "FHS" };
+
+        public JsonToHl7v2Processor(ProcessorSettings processorSettings)
+            : base(processorSettings)
+        {
+        }
+
+        public override string Convert(string data, string rootTemplate, ITemplateProvider templateProvider, TraceInfo traceInfo = null)
+        {
+            object jsonData;
+
+            jsonData = _parser.Parse(data);
+
+            var result = Convert(jsonData, rootTemplate, templateProvider, traceInfo);
+
+            var hl7Message = GenerateHL7Message(JObject.Parse(result));
+
+            var hl7String = ConvertHl7MessageToString(hl7Message);
+
+            return hl7String;
+        }
+
+        public string Convert(JObject data, string rootTemplate, ITemplateProvider templateProvider, TraceInfo traceInfo = null)
+        {
+            var jsonData = data.ToObject();
+            var result = Convert(jsonData, rootTemplate, templateProvider, traceInfo);
+            var hl7Message = GenerateHL7Message(JObject.Parse(result));
+
+            var hl7String = ConvertHl7MessageToString(hl7Message);
+            return hl7String;
+        }
+
+        public Hl7v2Data GenerateHL7Message(JObject transformations)
+        {
+            var hl7Segments = new List<Hl7v2Segment>();
+
+            // validate that the template has a messageDefinition section
+            if (transformations["messageDefinition"] == null)
+            {
+                throw new RenderException(
+                    FhirConverterErrorCode.TemplateRenderingError,
+                    "The messageDefinition section was not found in the conversion template");
+            }
+
+            // create HL7v2 segments
+            for (int i = 0; i < transformations["messageDefinition"].Count(); i++)
+            {
+                var dataResolution = transformations["messageDefinition"][i] as JObject;
+                var segmentName = dataResolution.Properties().First().Name;
+
+                var fields = new List<Hl7v2Field>();
+
+                JObject mappingsForSegment = dataResolution.Properties().First().Value as JObject;
+
+                // Fill in any gaps in fields for the given segment
+                //   - For example if PID-3 is populated but not PID-1 or PID-2 then we should fill in placeholders up until and including PID-3
+                //   - Get highest field position in each segment and fill in lower fields with blank fields
+                int highestField = 0;
+                foreach (var mapping in mappingsForSegment)
```

Comment:
Just for my learning, I will probably reach out to you offline to learn more about this part, in particular, I think an example of what mappingsForSegment looks like would be super helpful. 
Also, it's pretty cool that you were able to figure out Hl7v2 format in depth and even write a liquid template from scratch!